NLog.Web.AspNetCore/LayoutRenderers/AspNetRequestPostedBody.cs

```diff
@@ -47,75 +50,109 @@ protected override void DoAppend(StringBuilder builder, LogEventInfo logEvent)
             {
                 return;
             }
-
-            long? contentLength = httpRequest.ContentLength;
-            if (contentLength <= 0)
+          
+            if (!TryGetBody(httpRequest, out var body))
             {
                 return; // No Body to read
             }
 
-            if (MaxContentLength > 0 && contentLength > MaxContentLength)
+            // reset if possible
+            if (!TryResetStream(httpRequest, out var oldPosition, ref body))
```

Comment:
Think that TryGetBody() should be reponsible for returning a body where CanSeek is true. TryResetStream should not return a new body.

Reply:
Yes, that's is better indeed. Feel free to change 👼 

I'm off for today :)